### PR TITLE
Add input range kwarg for RescaleIntensity

### LINKED ISSUE #41: Rescale transform needs an input range

### PR DESCRIPTION
**Pull request by fepegar**
Fixes #41.

**Description**
It is sometimes useful to specify an input range that will be linearly mapped to the output range. For example, users might want to map specific Hounsfield Units to -1 and 1, making the normalization transform robust to intensity outliers.

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/master/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [x] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [x] In-line docstrings updated
- [x]  Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [ ] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Comment by fepegar** [comment]
```python
import torchio as tio
ct = tio.datasets.Slicer('CTChest')
```

Screen Shot 2021-04-08 at 19 55 45

```python
rescale = tio.RescaleIntensity(out_min_max=(-1, 1), in_min_max=(-1000, 1000))
rescale(ct).CT_chest.save('/tmp/rescaled.nii.gz')
```

Screen Shot 2021-04-08 at 19 57 45
